### ISSUE TITLE
TensorFlow error

### ISSUE THREAD

**Issue by kulichbulich** [RadeonOpenCompute/ROCm#1402]
Hello,
I have problem with training CNN in tensorflow-rocm-2.4.0. The error occurs randomly, sometimes after several iterations of learning CNN and sometimes it appears immediately. This error (complete error log in file **tensor-flow-rocm-error.txt**): 

_Fit:
2021-03-07 17:54:13.140698: I tensorflow/compiler/mlir/mlir_graph_optimization_pass.cc:116] None of the MLIR optimization passes are enabled (registered 2)
Epoch 1/200
2021-03-07 17:54:14.742459: I tensorflow/stream_executor/platform/default/dso_loader.cc:49] Successfully opened dynamic library libMIOpen.so
2021-03-07 17:54:14.889237: I tensorflow/stream_executor/platform/default/dso_loader.cc:49] Successfully opened dynamic library librocblas.so
89/89 [==============================] - 141s 2s/step - loss: nan - accuracy: 0.5137 - val_loss: nan - val_accuracy: 0.6555
Epoch 2/200
30/89 [=========>....................] - ETA: 24s - loss: nan - accuracy: 0.6652Memory access fault by GPU node-1 (Agent handle: 0x5f6d730) on address 0x7fb5d8712000. Reason: Page not present or supervisor privilege.
Aborted (core dumped)_

This is my configuration:
_Linux kulich-PC-GPU 5.4.0-66-generic #74~18.04.2-Ubuntu SMP Fri Feb 5 11:17:31 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux_
_RX Vega 64 8GB_

I try to reduce dataset size but the error still occurs. Where can be the problem?  Too complex model?

Another log file is included (**rocminfo.txt** and **dmesg.txt**)
Thank you for your work

[dmesg.txt](https://github.com/RadeonOpenCompute/ROCm/files/6097618/dmesg.txt)
[rocminfo.txt](https://github.com/RadeonOpenCompute/ROCm/files/6097619/rocminfo.txt)
[tensor-flow-rocm-error.txt](https://github.com/RadeonOpenCompute/ROCm/files/6097620/tensor-flow-rocm-error.txt)

**Comment by ROCmSupport** [comment]
Thanks @kulichbulich for reaching out.
Can you please share the exact steps(step by step) to reproduce the problem.
Thank you.

**Comment by aman190202** [comment]
Try filtering the dataset for any missing or invalid values. My model resulted in nan loss value whenever there was an invalid value in my dataset.

**Comment by kulichbulich** [comment]
Thank you All, for response.
I am creating Simple project here on [OneDriveStorage](https://1drv.ms/u/s!AhZEc7wU8vsTiDAV4iNffOHbn6ti?e=BH7ilQ) with a simple project (run: _python3 RoadTestUnet.py_).
To **aman190202** - I don't thing the problem is in dataset, because the same project was worked before update Rocm and tensorflow-cpu train network without problem, but is very slow.

**Comment by ROCmSupport** [comment]
Hi @kulichbulich 
Can you please share the exact steps to understand the problem clearly.
Thank you

**Comment by kulichbulich** [comment]
Please download 4files from  [OneDriveStorage](https://1drv.ms/u/s!AhZEc7wU8vsTiDAV4iNffOHbn6ti?e=BH7ilQ) to one directory and run training script _python3 RoadTestUnet.py_ . This script create model and load training file. During training error described on the top of page should appears.

**Comment by ROCmSupport** [comment]
Thanks @kulichbulich .
Let me take a look.

**Comment by ROCmSupport** [comment]
Hi @kulichbulich 
Please share the above details so that I will start working on it asap.
Thank you.

**Comment by kulichbulich** [comment]
After update to tensorflow-rocm-2.4.1 and ROCm 4.1, problem disappeared and training is working again now.
Thank you

**Comment by ROCmSupport** [comment]
Great to know that TF is working with ROCm 4.1 + TF2.4
Thank you
